### PR TITLE
Throw RangeError in case old style is mixed with new style.

### PR DESCRIPTION
**Pull request by zbraniecki**
Fixes #40.

### CODE REVIEW

**Comment by zbraniecki** [comment]
@FrankYFTang - wdyt?

**Comment by zbraniecki** [comment]
@FrankYFTang - I updated the patch to respect `hourCycle`. Quite frankly, I'm not sure anymore how the spec made sense before, because there's no place where `hourCycle` gets assigned to `hc` otherwise.

Anyhoo, now it should work. Can you review? This seems like a last remaining item before Stage 4!

File: spec.html [hunk 1]
```diff
@@ -144,6 +146,8 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
           1. <ins>Set _dateTimeFormat_.[[TimeStyle]] to _timeStyle_.</ins>
           1. <ins>Set _dateTimeFormat_.[[HourCycle]] to _hc_.</ins>
         1. <ins>If _dateStyle_ or _timeStyle_ are not *undefined*, then</ins>
+          1. <ins>If any of the Properties from <emu-xref href="#table-datetimeformat-components"></emu-xref> except of `hourCycle`, is defined in _options_,</ins>
```
Comment (ljharb) [inline spec.html]: “except of” doesn’t read right to me.
```suggestion
          1. <ins>If any of the Properties from <emu-xref href="#table-datetimeformat-components"></emu-xref>, not including `hourCycle`, is defined in _options_,</ins>
```

**Comment by anba** [comment]
> I updated the patch to respect `hourCycle`. Quite frankly, I'm not sure anymore how the spec made sense before, because there's no place where `hourCycle` gets assigned to `hc` otherwise.

The change for "Set hc to hourCycle." looks wrong to me, because ResolveLocale gets the "hourCycle" option through `opt` record and will then copy it into the `r` record.

**Comment by anba** [comment]
Furthermore the current patch will likely result in always throwing a RangeError, because `ToDateTimeOptions` fills in default values from <https://tc39.es/ecma402/#table-datetimeformat-components>. In other words, `ToDateTimeOptions` also needs to be updated.

**Comment by sffc** [comment]
@Ms2ger can you review this?

File: spec.html [hunk 2]
```diff
@@ -144,6 +146,8 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
           1. <ins>Set _dateTimeFormat_.[[TimeStyle]] to _timeStyle_.</ins>
           1. <ins>Set _dateTimeFormat_.[[HourCycle]] to _hc_.</ins>
         1. <ins>If _dateStyle_ or _timeStyle_ are not *undefined*, then</ins>
+          1. <ins>If any of the Properties from <emu-xref href="#table-datetimeformat-components"></emu-xref>, not including `hourCycle`, is defined in _options_,</ins>
```
Comment (Ms2ger) [inline spec.html]: What kind of value is _options_? Is it a JS object? Does this check imply `Get()` calls?

**Comment by anba** [comment]
I've opened #43, which supersedes this pull request.

**Comment by FrankYFTang** [comment]
> Furthermore the current patch will likely result in always throwing a RangeError, because `ToDateTimeOptions` fills in default values from https://tc39.es/ecma402/#table-datetimeformat-components. In other words, `ToDateTimeOptions` also needs to be updated.

I agree with @anba This spec change itself won't work. ToDateTimeOptions will cause it always throw RangeError

**Pull request closed without merging**